api/v3/Case.php

```diff
@@ -215,14 +215,18 @@ function _civicrm_api3_case_delete_spec(&$params) {
  *   'client_id' => finds all cases with a specific client
  *   'activity_id' => returns the case containing a specific activity
  *   'contact_id' => finds all cases associated with a contact (in any role, not just client)
+ * $params CRM_Utils_SQL_Select $sql
+ *   Other apis wishing to wrap & extend this one can pass in a $sql object with extra clauses
  *
  * @throws API_Exception
  * @return array
  *   (get mode, case_id provided): Array with case details, case roles, case activity ids, (search mode, case_id not provided): Array of cases found
  */
-function civicrm_api3_case_get($params) {
+function civicrm_api3_case_get($params, $sql = NULL) {
```

Comment:
Some observation while evaluating this change:

When `MagicFunctionProvider` calls this, [it theoretically could pass through `$extra`](https://github.com/civicrm/civicrm-core/blob/4.7.17/Civi/API/Provider/MagicFunctionProvider.php#L89) which would break this contract. I've never really understood `$extra`, but here's a fun fact: MagicFunctionProvider probably broke `$extra` since `$extra` isn't actually inputted there. (It should be `$apiRequest['extra']`.) 

Upshot: If `$extra` has been broken all this time, and it doesn't serve any clear purpose, then perhaps we can rip it out...